app/models/user_access.rb

```diff
@@ -133,6 +156,48 @@ def can_access_record?(resources, record)
     resources.exists?
   end
 
+  def can_act_on_facility?(action, operation, record)
```

Comment:
Here we can do the appropriate checks for various operations based on the model and its nuances